barman/backup_executor.py

```diff
@@ -1588,6 +1628,40 @@ def stop_backup(self, backup_info):
             # Skip switching XLOG if a standby server
             pass
 
+    def _read_compressed_backup_label(self, backup_info):
```

Comment:
This bothers me because we must scan the tarball in order to find the backup_label - as it happens, the backup label will be one of the first members however this is only by luck and is not a documented behaviour that we should be relying on. If it turned out that backup_label was later in the tarball then Barman could be waiting a long time while we're extracting it.

Reply:
Does it actually take more time to get the latest file in the archive than the first one using `TarFile.open`?  I was guessing it would behave like `tar -tvf file.tar` that list files to get the expected file, then extract the file. In wich case the only difference regarding file position would be the file list size (that would be max O(n) depending on the search algorithm). I admit that it could still take a while for a large backup.

Reply:
Yes it is just a single scan of the tarfile, since once we have the member the subsequent extract does not require another scan - but indeed with a large enough backup that could take a long time.

Perhaps it's a case of "but a small enough overhead compared to the time taken to copy the file over the network" though.